orte/mca/ras/base/ras_base_node.c

```diff
@@ -157,10 +157,12 @@ int orte_ras_base_node_insert(opal_list_t* nodes, orte_job_t *jdata)
                     opal_argv_free(nalias);
                 }
                 /* and store the result */
-                ptr = opal_argv_join(alias, ',');
+                if (0 < opal_argv_count(alias)) {
+                    ptr = opal_argv_join(alias, ',');
+                    orte_set_attribute(&hnp_node->attributes, ORTE_NODE_ALIAS, ORTE_ATTR_LOCAL, ptr, OPAL_STRING);
+                    free(ptr);
+                }
```

Comment:
Why split the original value that you just obtained from `orte_get_attribute(..., ORTE_NODE_ALIAS, ...)`?  If you get a non-NULL/non-empty value back, why not just directly `orte_set_attribute(...)`?

I ask because you're not saving the value you get back from `orte_get_attribute()` -- you're getting the value from `orte_get_attribute()`, transforming it, then transforming it back, and then using the double-transformed value to call `orte_set_attribute()`.  Am I missing something?

Reply:
We have to add the new alias to the end of the existing comma-delimited list, so we have to first fetch what we may already have, split it to get an argv array, then add only if unique, recombine and re-store. If he gets nothing back from the original "get", then he still has to store the new one.

Note that the code block 151-158 should be removed - looks like a copy/paste error.

Reply:
Ok. 👍

Reply:
@rhc54 at line 143, we get the alias(es) for `&hnp_node->attributes` but at line 151, we get the alias(es) for `&node->attributes`, so are you sure this block should be removed ?

note I will add a missing `free(ptr)` at line 153

also, what is the rationale for testing `ptr != NULL` at line 144 when we check the return of `orte_get_attribute()` everywhere else.

Reply:
@ggouaillardet You are right about the code block - I forgot that "node" may be carrying its own list of aliases.

Technically, it would probably be better to check the return code instead of checking ptr for NULL. Ultimately, it doesn't matter as ptr will obviously remain NULL if the attribute isn't found.